front/ui/ui-charts/src/manchette/hooks/useSyncManchette.ts

```diff
@@ -0,0 +1,186 @@
+    yZoom: 1,
+    timeOrigin: defaultTimeOrigin,
+    spaceOrigin: defaultSpaceOrigin,
+    xOffset: defaultXOffset,
+    yOffset: 0,
+    scrollTo: null,
+    panning: null,
+    zoomMode: false,
+    rect: null,
+    pixelRect: null,
+    scales: [],
+  });
+
+  const { yZoom, yOffset, scrollTo, rect } = state;
+
+  // =========== Shift pressed ===========
+  const handleKeyDown = useCallback((event: KeyboardEvent) => {
+    if (event.key === 'Shift') {
+      setIsShiftPressed(true);
+    }
+  }, []);
+
+  const handleKeyUp = useCallback((event: KeyboardEvent) => {
+    if (event.key === 'Shift') {
+      setIsShiftPressed(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    window.addEventListener('keydown', handleKeyDown);
+    window.addEventListener('keyup', handleKeyUp);
+
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+      window.removeEventListener('keyup', handleKeyUp);
+    };
+  }, [handleKeyDown, handleKeyUp]);
+
+  // =========== Y Zoom ===========
+  const zoomYIn = useCallback(() => {
+    const newYZoom = Math.min(yZoom + ZOOM_Y_DELTA, maxYZoom);
+    if (newYZoom !== yZoom) {
+      const newYOffset = yOffset * (newYZoom / yZoom);
+
+      setState((prev) => ({
+        ...prev,
+        yZoom: newYZoom,
+        yOffset: newYOffset,
+        scrollTo: newYOffset,
+      }));
+    }
+  }, [yZoom, maxYZoom, yOffset]);
+
+  const zoomYOut = useCallback(() => {
+    const newYZoom = Math.max(MIN_ZOOM_Y, yZoom - ZOOM_Y_DELTA);
+    if (newYZoom !== yZoom) {
+      const newYOffset = yOffset * (newYZoom / yZoom);
+      setState((prev) => ({
+        ...prev,
+        yZoom: newYZoom,
+        yOffset: newYOffset,
+        scrollTo: newYOffset,
+      }));
+    }
+  }, [yZoom, yOffset]);
+
+  const resetYZoom = useCallback(() => {
+    setState((prev) => ({ ...prev, yZoom: 1 }));
+  }, []);
+
+  // =========== X Zoom ===========
+  const handleXZoom = useCallback(
+    (newXZoom: number, xPosition = (diagramRef?.current?.offsetWidth || 0) / 2) => {
+      if (!options || options.enableTimeZoom)
+        setState((prev) => ({
+          ...prev,
+          ...zoomX(prev.xZoom, prev.xOffset, newXZoom, xPosition),
+        }));
+    },
+    [options, diagramRef]
+  );
+
+  // =========== Y Scroll ===========
+  const handleScrollInManchette = useCallback(
+    (e: React.UIEvent<HTMLDivElement>) => {
+      if (rect) {
+        e.preventDefault();
+        return;
+      }
+      if (!isShiftPressed && manchetteWithSpaceTimeChartRef.current) {
+        const { scrollTop } = manchetteWithSpaceTimeChartRef.current;
+        if (scrollTop || scrollTop === 0) {
+          setState((prev) => ({ ...prev, yOffset: scrollTop }));
+        }
+      }
+    },
+    [isShiftPressed, manchetteWithSpaceTimeChartRef, rect]
+  );
+
+  useEffect(() => {
+    if (scrollTo !== null && manchetteWithSpaceTimeChartRef.current) {
+      manchetteWithSpaceTimeChartRef.current.scrollTo({
```

Comment:
Orthogonal comment: I'm not sure why we use a `useEffect()` for this, instead of introducing a function that performs the `scrollTo()` that we can call directly instead of setting some state.

But the old code was structured this way too, so let's leave this alone for now.